### PR TITLE
feat(core22): maven plugin

### PR DESCRIPTION
**Pull request by cmatsuoka**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox run -m lint`?
- [ ] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

### CODE REVIEW

File: tests/spread/plugins/craft-parts/build-and-run-hello/maven-hello/snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+base: core22
+
+lint:
+  ignore:
+    - library:
+        - usr/lib/jvm/java-*/lib/*.so
+
+apps:
+  maven-hello:
+    command: bin/java -jar $SNAP/jar/hello-$SNAPCRAFT_PROJECT_VERSION.jar
```
Comment (tigarmo) [inline tests/spread/plugins/craft-parts/build-and-run-hello/maven-hello/snap/snapcraft.yaml:18]:
```suggestion
    command: bin/java -jar $SNAP/jar/hello-$CRAFT_PROJECT_VERSION.jar
```

**Review by tigarmo — CHANGES_REQUESTED** [review]
Looks good. We should also add a snapcraft-specific example (the one currently in craft-parts)

**Review by syu-w — APPROVED** [review]

**Review by mattculler — APPROVED** [review]

**Review by tigarmo — COMMENTED** [review]
The build is failing because that `/common/craft-parts/mave_plugin.rst` shared doc doesn't exist on that version of craft-parts. We should use the dev branch just to ensure the build is successful

File: requirements-devel.txt [hunk 2]
```diff
@@ -15,7 +15,8 @@ craft-application==1.2.1
 craft-archives==1.1.3
 craft-cli==2.5.1
 craft-grammar==1.1.2
-craft-parts==1.27.0
+#craft-parts==1.27.0
+craft-parts @ git+https://github.com/canonical/craft-parts@CRAFT-2584-Document-maven-plugin
```
Comment (cmatsuoka) [inline requirements-devel.txt:19]: Revert before landing.

**Review by tigarmo — APPROVED** [review]
thanks!

**Review by mattculler — APPROVED** [review]

**Comment by sergiusens** [comment]
tests are broken

**Pull request merged**